operator/cmd/mesh/manifest-generate_test.go

```diff
@@ -703,3 +708,281 @@ func generateManifest(inFile, flags string, chartSource chartSourceType) (string
+				{canaryRevLabel, empty, ""},
+				{legacyLabel, empty, "istiod"},
+				{legacyAndDefaultLabel, empty, "istiod"},
+
+				{legacyAndCanaryLabel, empty, "istiod"},
+			},
+		},
+		// We only have a revision label. We should match only on explicit revision label and send to that revision
+		{
+			name:     "revision only",
+			webhooks: revWebhook,
+			checks: []test{
+				{empty, empty, ""},
+				{defaultRevLabel, empty, ""},
+				{canaryRevLabel, empty, "istiod-canary"},
+				{legacyLabel, empty, ""},
+				{legacyAndDefaultLabel, empty, ""},
+				{legacyAndCanaryLabel, empty, ""},
+			},
+		},
+
+		// We have a revision and a default install
+		{
+			name:     "both",
+			webhooks: bothWebhooks,
+			checks: []test{
+				{empty, empty, ""},
+				{defaultRevLabel, empty, ""},
+				{canaryRevLabel, empty, "istiod-canary"},
+				{legacyLabel, empty, "istiod"},
+				{legacyAndDefaultLabel, empty, "istiod"},
+				{legacyAndCanaryLabel, empty, "istiod"},
+			},
+		},
+
+		// object selector enabled
+		// This should have the same behavior as before, but if the disable label is set its always disabled
+		{
+			name:     "object selector",
+			webhooks: objectSelectorWebhook,
+			checks: []test{
+				// no label
+				{empty, empty, ""},
+				{defaultRevLabel, empty, ""},
+				{canaryRevLabel, empty, ""},
+				{legacyLabel, empty, "istiod"},
+				{legacyAndDefaultLabel, empty, "istiod"},
+				{legacyAndCanaryLabel, empty, "istiod"},
+
+				// disabled
+				{empty, objDisable, ""},
+				{defaultRevLabel, objDisable, ""},
+				{canaryRevLabel, objDisable, ""},
+				{legacyLabel, objDisable, ""},
+				{legacyAndDefaultLabel, objDisable, ""},
+				{legacyAndCanaryLabel, objDisable, ""},
+
+				// enabled
+				{empty, objEnabled, ""},
+				{defaultRevLabel, objEnabled, ""},
+				{canaryRevLabel, objEnabled, ""},
+				{legacyLabel, objEnabled, "istiod"},
+				{legacyAndDefaultLabel, objEnabled, "istiod"},
+				{legacyAndCanaryLabel, objEnabled, "istiod"},
+			},
+		},
+
+		// object selector enabled, with a revision
+		{
+			name:     "object selector revision",
+			webhooks: objectSelectorWebhookRevision,
+			checks: []test{
+				{empty, objDisable, ""},
+				{defaultRevLabel, objDisable, ""},
+				{canaryRevLabel, objDisable, ""},
+				{legacyLabel, objDisable, ""},
+				{legacyAndDefaultLabel, objDisable, ""},
+				{legacyAndCanaryLabel, objDisable, ""},
+
+				{empty, objEnabled, ""},
+				{defaultRevLabel, objEnabled, ""},
+				{canaryRevLabel, objEnabled, "istiod-canary"},
+				{legacyLabel, objEnabled, ""},
+				{legacyAndDefaultLabel, objEnabled, ""},
+				{legacyAndCanaryLabel, objEnabled, ""},
+			},
+		},
+
+		// Auto injection
+		{
+			name:     "auto",
+			webhooks: autoWebhook,
+			checks: []test{
+				{empty, empty, "istiod"},
+				{canaryRevLabel, empty, ""},
+				{legacyLabel, empty, "istiod"},
+				{legacyLabelDisabled, empty, ""},
+				// TODO: This is sort of a weird behavior. We should still be able to explicitly select the revision
```

Comment:
is there a doc that documents all of these behaviors, specifically combination with auto injection case, for some cases it it not really intuitive

Reply:
No, it's undocumented alpha experimental behavior. I am just trying to capture the current state so we can improve on it